.github/workflows/mikeals-workflow.yml

```diff
@@ -22,6 +22,8 @@ jobs:
     - name: Build
       if: steps.cache-modules.outputs.cache-hit != 'true'
       run: npm install
+    - name: Typecheck
```

Comment:
We shouldn’t modify this file directly.

The reason it’s specifically named “mikeals-workflow.yml” is because I have a script that looks at all the repos in all of the orgs i’m in and  updates this file when I have changes.

But more importantly, I don’t want to impose any knowledge of Typescript on contributors who just want to contribute JavaScript code to the project. The types will get built as part of the test process, so if any code goes in that causes a regression in the type generation we’ll find it. But running this full type check may throw exceptions on things that aren’t actually problems in JavaScript (looks like you’ve actually found a bunch below already).

Contributors who are familiar with Typescript will need to be responsible for maintaining this compatibility. I’m not willing to impose this maintenance burden on every potential contributor.